--- init	2019-11-06 05:48:06.000000000 +0800
+++ abc/init_new	2020-12-24 11:59:58.000000000 +0800
@@ -59,7 +59,7 @@
 
 echo "L4T-INITRD Build DATE: $build_date" > /dev/kmsg;
 
-rootdev="$(sed -ne 's/.*\broot=\/dev\/\([abcdefklmnps0-9]*\)\b.*/\1/p' < /proc/cmdline)"
+rootdev="$(sed -ne 's/.*\broot=\/dev\/\([abcdefklmnpsv0-9]*\)\b.*/\1/p' < /proc/cmdline)"
 if [ "${rootdev}" == "" ]; then
 	uuid_regex='[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}'
 	rootdev="$(sed -ne "s/.*\broot=\(PARTUUID=${uuid_regex}\)\b.*/\1/p" < /proc/cmdline)"
@@ -85,7 +85,7 @@
 		echo "ERROR: ${rootdev} mount fail..." > /dev/kmsg;
 		exec /bin/bash;
 	fi;
-elif [[ "${rootdev}" == mmcblk* ]]; then
+elif [[ "${rootdev}" == mmcblk* ]] || [[ "${rootdev}" == nvme0n* ]]; then
 	if [ ! -e "/dev/${rootdev}" ]; then
 		count=0;
 		while [ ${count} -lt 50 ]
